### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by snapshotpl**
Remove static, factories, remove extend in filter, extends service providers and more test

### CODE REVIEW

File: src/BaconStringUtils/Filter/Slugify.php [hunk 1]
```diff
@@ -15,13 +15,34 @@
...
+
+    /**
+     * @param Slugifier $slugifier
+     */
+    public function setSlugifier(Slugifier $slugifier)
+    {
+        $this->slugifier = $slugifier;
+    }
+
+    /**
```
Comment (DASPRiD) [inline src/BaconStringUtils/Filter/Slugify.php:41]: I'm actually wondering… do we need this getter at all?

File: src/BaconStringUtils/Slugifier.php [hunk 2]
```diff
@@ -36,7 +30,11 @@ class Slugifier
      */
     public function slugify($string)
     {
-        $string = $this->uniDecoder()->decode($string);
+        $decoder = $this->getUniDecoder();
+
+        if ($decoder instanceof UniDecoder) {
```
Comment (DASPRiD) [inline src/BaconStringUtils/Slugifier.php:35]: I'd argue that this is a hard dependency. Either let the getUniDecoder() instantiate it if not set or require it in the constructor and remove the setter and getter.

Comment (snapshotpl) [inline src/BaconStringUtils/Slugifier.php:35]: Slugifier still works great without decoder

Comment (DASPRiD) [inline src/BaconStringUtils/Slugifier.php:35]: Mh, good point.

File: src/BaconStringUtils/Filter/Slugify.php (hunk 1 above)
Comment (snapshotpl) [inline src/BaconStringUtils/Filter/Slugify.php:41]: In module we don't use it, but someone want to get it.

Comment (DASPRiD) [inline src/BaconStringUtils/Filter/Slugify.php:41]: I don't really see any reason for it. After all, they may get it through the service manager.

**Comment by DASPRiD** [comment]
Also, make sure to fix the build.

File: src/BaconStringUtils/Filter/Slugify.php (hunk 1 above)
Comment (snapshotpl) [inline src/BaconStringUtils/Filter/Slugify.php:41]: What if you want to check slugifier in filter? You don't see reason to keep it, I don't see reason to remote it.

**Comment by snapshotpl** [comment]
@DASPRiD Travis fail on a test. Can you restart it manually?

**Comment by DASPRiD** [comment]
Actually PHPCS fails.

**Comment by snapshotpl** [comment]
I have pushed new commit 'Fix coding standard', but test failed on composer

**Comment by DASPRiD** [comment]
I see. Well, I cannot restart travis on pull requests.

**Comment by snapshotpl** [comment]
@DASPRiD Green light :-)

**Comment by snapshotpl** [comment]
@DASPRiD should I fix something more?

File: src/BaconStringUtils/Filter/SlugifyFactory.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * BaconStringUtils
+ *
+ * @link      http://github.com/Bacon/BaconStringUtils For the canonical source repository
+ * @copyright 2011-2014 Ben Scholzen 'DASPRiD'
+ * @author    Witold Wasiczko <witold@wasiczko.pl>
```
Comment (DASPRiD) [inline src/BaconStringUtils/Filter/SlugifyFactory.php:7]: We don't use author tags :)

File: src/BaconStringUtils/SlugifierFactory.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * BaconStringUtils
+ *
+ * @link      http://github.com/Bacon/BaconStringUtils For the canonical source repository
+ * @copyright 2011-2014 Ben Scholzen 'DASPRiD'
+ * @author Witold Wasiczko <witold@wasiczko.pl>
```
Comment (DASPRiD) [inline src/BaconStringUtils/SlugifierFactory.php:7]: Same

File: tests/BaconStringUtils/Filter/SlugifyFactoryTest.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * BaconStringUtils
+ *
+ * @link      http://github.com/Bacon/BaconStringUtils For the canonical source repository
+ * @copyright 2011-2014 Ben Scholzen 'DASPRiD'
+ * @author    Witold Wasiczko <witold@wasiczko.pl>
```
Comment (DASPRiD) [inline tests/BaconStringUtils/Filter/SlugifyFactoryTest.php:7]: Same

**Pull request merged**